src/System.Diagnostics.DiagnosticSource/tests/DiagnosticSourceEventSourceBridgeTests.cs

```diff
@@ -427,6 +427,76 @@ public void TestMessages()
+                Assert.Equal(1, eventSourceListener.EventCount); // Exactly one more event has been emitted.
+                Assert.Equal("Activity2Start", eventSourceListener.LastEvent.EventSourceEventName);
+                Assert.Equal("TestActivitiesSource", eventSourceListener.LastEvent.SourceName);
+                Assert.Equal("TestActivity2Start", eventSourceListener.LastEvent.EventName);
+                Assert.Equal(1, eventSourceListener.LastEvent.Arguments.Count);
+                Assert.Equal("start", eventSourceListener.LastEvent.Arguments["propStr"]);
+                eventSourceListener.ResetEventCountAndLastEvent();
+
+                // Send a normal event 
+                diagnosticSourceListener.Write("TestEvent", new { propStr = "event" });
+                Assert.Equal(1, eventSourceListener.EventCount); // Exactly one more event has been emitted.
+                Assert.Equal("Event", eventSourceListener.LastEvent.EventSourceEventName);
+                Assert.Equal("TestActivitiesSource", eventSourceListener.LastEvent.SourceName);
+                Assert.Equal("TestEvent", eventSourceListener.LastEvent.EventName);
+                Assert.Equal(1, eventSourceListener.LastEvent.Arguments.Count);
+                Assert.Equal("event", eventSourceListener.LastEvent.Arguments["propStr"]);
+                eventSourceListener.ResetEventCountAndLastEvent();
+
+                // Stop nested activity 2
+                diagnosticSourceListener.Write("TestActivity2Stop", new { propStr = "stop" });
+                Assert.Equal(1, eventSourceListener.EventCount); // Exactly one more event has been emitted.
+                Assert.Equal("Activity2Stop", eventSourceListener.LastEvent.EventSourceEventName);
+                Assert.Equal("TestActivitiesSource", eventSourceListener.LastEvent.SourceName);
+                Assert.Equal("TestActivity2Stop", eventSourceListener.LastEvent.EventName);
+                Assert.Equal(1, eventSourceListener.LastEvent.Arguments.Count);
+                Assert.Equal("stop", eventSourceListener.LastEvent.Arguments["propStr"]);
+                eventSourceListener.ResetEventCountAndLastEvent();
+
+                // Stop activity 1
+                diagnosticSourceListener.Write("TestActivity1Stop", new { propStr = "stop" });
+                Assert.Equal(1, eventSourceListener.EventCount); // Exactly one more event has been emitted.
+                Assert.Equal("Activity1Stop", eventSourceListener.LastEvent.EventSourceEventName);
+                Assert.Equal("TestActivitiesSource", eventSourceListener.LastEvent.SourceName);
+                Assert.Equal("TestActivity1Stop", eventSourceListener.LastEvent.EventName);
+                Assert.Equal(1, eventSourceListener.LastEvent.Arguments.Count);
+                Assert.Equal("stop", eventSourceListener.LastEvent.Arguments["propStr"]);
+                eventSourceListener.ResetEventCountAndLastEvent();
```

Comment:
Adding extra harnessing (basically abstractions that do 'routine tests' can have downsides, as it makes the tests less approachable (and less useful as 'sample' / documentation).   I agree that after a point it would be worth it, but I don't think we are there yet...